src/index.js

```diff
@@ -1,5 +1,5 @@
 export { Trans } from './Trans.js';
-export { Trans as TransWithoutContext } from './TransWithoutContext.js';
+export { Trans as TransWithoutContext } from './TransWithoutContext.jsx';
```

Comment:
can we keep this js instead of jsx?

Reply:
> TransWithoutContext file have been renamed with jsx extension otherwise vitest is not able to compile it correctly.

`vitest` complains when compiling a js file which contains actual jsx (rightfully IMHO 😁)

<img width="1159" alt="image" src="https://github.com/i18next/react-i18next/assets/24919330/614151b7-e37b-42b4-ba5b-34b229d65799">

----

> Regular build should not be affect by this change.

I have executed `npm run build` and dist files are not modified so I don't think that this change will affect library usage

Reply:
I don't think the dist files are not modified... for sure the path to .tsx is changed...
I like to keep .js if possible

Reply:
maybe we can remove the jsx parts:

<img width="1109" alt="image" src="https://github.com/i18next/react-i18next/assets/1086194/83606167-4dbf-4669-a700-a8ba61edefad">
<img width="1118" alt="image" src="https://github.com/i18next/react-i18next/assets/1086194/a041a478-85f4-40ff-8ed7-1818a2bd7c0c">

Reply:
```js
import { isValidElement, cloneElement, createElement, Children, Fragment } from 'react';
// ...
          return createElement(c.type, { ...props, key: i, ref: c.ref, ...(isVoid ? {} : { children: inner }) });
          // return <c.type {...props} key={i} ref={c.ref} {...(isVoid ? {} : { children: inner })} />;
// ...
      function Componentized() {
        return createElement(Fragment, null, comp);
        // return <>{comp}</>;
      }
      components[c] = createElement(Componentized, null);
      // components[c] = <Componentized />;
```

Reply:
> I don't think the dist files are not modified... for sure the path to .jsx is changed...
> I like to keep .js if possible

`dist` files have always `js` extension because they are compiled by `rollup`. 
You can't have jsx file inside dist. (Try run the build on `master` and this PR branch)

> maybe we can remove the jsx parts:

Sure, I can make this change.